src/main/java/io/rubuy74/rhs/adapter/out/messaging/KafkaMarketChangePublisher.java

```diff
@@ -0,0 +1,47 @@
+package io.rubuy74.rhs.adapter.out.messaging;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.rubuy74.rhs.converter.serialization.MarketOperationSerializer;
+import io.rubuy74.rhs.domain.MarketOperation;
+import io.rubuy74.rhs.port.out.MarketChangePublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class KafkaMarketChangePublisher implements MarketChangePublisher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaMarketChangePublisher.class);
+    private static final String TOPIC = "market-changes";
+    private final KafkaTemplate<String, byte[]> kafkaTemplate;
+
+    public KafkaMarketChangePublisher(ObjectMapper objectMapper,KafkaTemplate<String, byte[]> kafkaTemplate) {
+        this.kafkaTemplate = kafkaTemplate;
+    }
+
+    @Override
+    public void publish(MarketOperation marketOperation) {
+        byte[] payload = MarketOperationSerializer.serialize(marketOperation);
+        try {
+            kafkaTemplate.send(TOPIC, payload)
+                .whenComplete((result, e) -> {
+                    if (e != null) {
+                        LOGGER.error("operation=publish_market_operation," +
```

Comment:
we always mark the operation as the method name for easier tracing

```suggestion
                        LOGGER.error("operation=publish," +
```